libp2p/Host.cpp

```diff
@@ -137,27 +145,24 @@ void Host::doneWorking()
 	m_ioService.reset();
 	
 	// finally, clear out peers (in case they're lingering)
-	RecursiveGuard l(x_peers);
-	m_peers.clear();
+	RecursiveGuard l(x_sessions);
+	m_sessions.clear();
 }
 
 unsigned Host::protocolVersion() const
 {
-	return 2;
+	return 3;
 }
 
 void Host::registerPeer(std::shared_ptr<Session> _s, CapDescs const& _caps)
 {
-	if (!_s->m_node || !_s->m_node->id)
 	{
-		cwarn << "Attempting to register a peer without node information!";
-		return;
-	}
-
-	{
-		RecursiveGuard l(x_peers);
-		m_peers[_s->m_node->id] = _s;
+		RecursiveGuard l(x_sessions);
+		if (!m_peers.count(_s->m_peer->id))
+			m_peers[_s->m_peer->id] = _s->m_peer;
+		m_sessions[_s->m_peer->id] = _s;
```

Comment:
TL;DR -- Temporary; will be fixed w/tomorrow's PR. This is the middle-point of integration; post-integration, a session will not be created unless the peer exists. 
In this PR, m_peers will act as a storage backend. Pruning peers will come later and require a mutex in connect and disconnect events; for now we just do this when Host is deallocated. Even better, when m_peers is a first-class storage backend, it'll take care of itself.